src/id/mangatale/src/eu/kanade/tachiyomi/extension/id/mangatale/Ikiru.kt

```diff
@@ -1,42 +1,241 @@
 package eu.kanade.tachiyomi.extension.id.mangatale
 
-import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import android.content.SharedPreferences
+import android.util.Log
+import android.widget.Toast
+import androidx.preference.EditTextPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
-import okhttp3.MediaType.Companion.toMediaType
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.getPreferences
+import keiyoushi.utils.tryParse
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MultipartBody
 import okhttp3.OkHttpClient
-import okhttp3.ResponseBody.Companion.toResponseBody
+import okhttp3.Request
+import okhttp3.RequestBody
 import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
 
-class Ikiru : MangaThemesia("Ikiru", "https://id.ikiru.wtf", "id") {
-
+class Ikiru() : ParsedHttpSource(), ConfigurableSource {
+    // Formerly "MangaTale"
     override val id = 1532456597012176985
 
+    override val name = "Ikiru"
+    private val defaultBaseUrl: String = "https://01.ikiru.wtf"
+    override val lang = "id"
+    override val supportsLatest = true
+
+    override val baseUrl: String get() = when {
+        isCi -> defaultBaseUrl
+        else -> preferences.prefBaseUrl
+    }
```

Comment:
always return `preferences.prefBaseUrl`
```suggestion
    override val baseUrl: String get() = preferences.prefBaseUrl
```

Reply:
I was also wondering why the other ext also use that

Reply:
> I was also wondering why the other ext also use that

https://github.com/keiyoushi/extensions-source/pull/9651#discussion_r2205122173

Quoting my reply:

> The CI workflow can properly handle preferences.
> 
> This pattern is also misused in the repo. I wrote it to handle dynamic pulling URL from network on CI build, not for skipping preferences.